lib/gpio/pin.toit

```diff
@@ -387,6 +515,132 @@ class InvertedPin extends Pin:
   set-open-drain value/bool:
     original-pin_.set-open-drain value
 
+/**
+A GPIO chip on Linux.
+*/
+class Chip:
+  static resource-group_ ::= gpio-linux-chip-init_
+
+  /**
+  The path to the GPIO chip.
+  */
+  path/string
+  resource_/ByteArray? := ?
+  name_/string? := null
+  label_/string? := null
+  line-count_/int? := null
+
+  constructor .path:
+    resource_ = gpio-linux-chip-new_ resource-group_ path
+    add-finalizer this:: close
+
+  /**
+  Lists all GPIO chips on the system.
+
+  The returned list contains paths to the GPIO chips.
+  */
+  static list -> List:
+    return gpio-linux-list-chips_
+
+  /** The name of the GPIO chip. */
+  name -> string:
+    if not name_: fetch-info_
+    return name_
+
+  /** The label of the GPIO chip. */
+  label -> string:
+    if not label_: fetch-info_
+    return label_
+
+  /** The number of pins on the GPIO chip. */
+  pin-count -> int:
+    if not line-count_: fetch-info_
+    return line-count_
+
+  fetch-info_ -> none:
+    info/List := gpio-linux-chip-info_ resource_
+    name_ = info[0]
+    label_ = info[1]
+    line-count_ = info[2]
+
+  pin-names -> List:
+    return List pin-count:
+      info := gpio-linux-chip-pin-info_ resource_ it
+      info[0]
+
+  /**
+  Closes the chip.
+
+  If children (like pins) are still open, the closing of the chip is delayed
+    until all children are closed.
+  */
+  close:
+    if resource_:
+      resource := resource_
+      resource_ = null
```

Comment:
Once you've done this, you might as well just remove the finalizer before calling pin-close. The finalizer will not do anything. This way, you can avoid the try-finally.

Reply:
done.